libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -337,7 +335,44 @@ private void syncUp(SyncState sync, SyncUpdateCallback callback) throws Exceptio
         }
 	}
 
-    private boolean syncUpOneRecord(String soupName, List<String> fieldlist, JSONObject record) throws JSONException, IOException {
+    private boolean isNewerThanServer(String objectType, String objectId,
+    		long lastModifiedDate) throws JSONException, IOException {
+    	boolean isNewer = false;
+    	long serverLastModified = UNCHANGED;
+    	final SOQLBuilder builder = SOQLBuilder.getInstanceWithFields(Constants.LAST_MODIFIED_DATE);
+        builder.from(objectType);
+        builder.where(Constants.ID + " = '" + objectId + "'");
+        final String query = builder.build();
+        RestResponse lastModResponse = null;
+        lastModResponse = sendSyncWithSmartSyncUserAgent(RestRequest.getRequestForQuery(apiVersion, query));
```

Comment:
This will always occur within the context of an asynchronous transaction, yes?  Otherwise, a sync request here could be problematic....